homeassistant/components/device_tracker/owntracks.py

```diff
@@ -196,70 +201,70 @@ def owntracks_event_update(topic, payload, qos):
-                if location in regions:
-                    regions.remove(location)
-                new_region = regions[-1] if regions else None
-
-                if new_region:
-                    # Exit to previous region
-                    zone = hass.states.get(
-                        "zone.{}".format(slugify(new_region)))
-                    _set_gps_from_zone(kwargs, new_region, zone)
-                    _LOGGER.info("Exit to %s", new_region)
-                    see(**kwargs)
-                    see_beacons(dev_id, kwargs)
-
-                else:
-                    _LOGGER.info("Exit to GPS")
-                    # Check for GPS accuracy
-                    valid_gps = True
-                    if 'acc' in data:
-                        if data['acc'] == 0.0:
-                            valid_gps = False
-                            _LOGGER.warning(
-                                'Ignoring GPS in region exit because accuracy'
-                                'is zero: %s',
-                                payload)
-                        if (max_gps_accuracy is not None and
-                                data['acc'] > max_gps_accuracy):
-                            valid_gps = False
-                            _LOGGER.info(
-                                'Ignoring GPS in region exit because expected '
-                                'GPS accuracy %s is not met: %s',
-                                max_gps_accuracy, payload)
-                    if valid_gps:
-                        see(**kwargs)
-                        see_beacons(dev_id, kwargs)
-
-                beacons = MOBILE_BEACONS_ACTIVE[dev_id]
-                if location in beacons:
-                    beacons.remove(location)
-                    _LOGGER.info("Remove beacon %s", location)
+            regions = regions_entered[dev_id]
+            if location in regions:
+                regions.remove(location)
+            new_region = regions[-1] if regions else None
+
+            if new_region:
+                # Exit to previous region
+                zone = hass.states.get(
+                    "zone.{}".format(slugify(new_region)))
+                _set_gps_from_zone(kwargs, new_region, zone)
+                _LOGGER.info("Exit to %s", new_region)
+                hass.async_add_job(async_see(**kwargs))
+                async_see_beacons(dev_id, kwargs)
+
+            else:
+                _LOGGER.info("Exit to GPS")
+                # Check for GPS accuracy
+                valid_gps = True
+                if 'acc' in data:
+                    if data['acc'] == 0.0:
+                        valid_gps = False
+                        _LOGGER.warning(
+                            'Ignoring GPS in region exit because accuracy'
+                            'is zero: %s',
+                            payload)
+                    if (max_gps_accuracy is not None and
+                            data['acc'] > max_gps_accuracy):
```

Comment:
Just curious, how is `is not None` "just a truthy check"? If `max_gps_accuracy` is 0, then the two statements are **not** equivalent (unless this behavior is different in Python 3).

Now, I understand that setting it to 0 is probably an unlikely scenario since that would essentially ignore everything except zone transitions, but it's technically a possible use case currently and changing this would mean that's no longer allowed because a `max_gps_accuracy` of 0 would be ignored. You could still work around it by using a really small, non-zero value, such as 0.000001.

I don't really care either way, I'm not setting this to 0 so it doesn't affect me. I just wanted to make sure I'm not misunderstanding something here.